mmdeploy/codebase/mmseg/models/segmentors/encoder_decoder.py

```diff
@@ -25,6 +25,14 @@ def encoder_decoder__predict(self, inputs, data_samples, **kwargs):
         batch_img_metas.append(data_sample.metainfo)
     x = self.extract_feat(inputs)
     seg_logit = self.decode_head.predict(x, batch_img_metas, self.test_cfg)
+
+    # mark seg_head
+    @mark('decode_head', outputs=['output'])
+    def __mark_seg_logit(seg_logit):
+        return seg_logit
+
+    seg_logit = __mark_seg_logit(seg_logit)
```

Comment:
If vacc does not support argmax, `End2EndModel` in `mmdeploy/codebase/mmseg/deploy/segmentation_model.py` may raise error. You may update `End2EndModel` or you can register a new class like `RKNNModel`.

Reply:
I implement `VACCModel`， but when i test, still set `End2EndModel` type. so there maybe a bug in segmentation wrapper and i delete the implementation. can you tell me except the `VACCModel` function how can get the `VACCModel`, the config have setted
```json
partition_config = dict(
    type='vacc',
    apply_marks=True,
    partition_cfg=[
        dict(
            save_file='model.onnx',
            start=['segmentor_forward:output'],
            end=['decode_head:input'],
            output_names=['feat'])
    ])
```

Reply:
Compare with detection task, `build_object_detection_model` function set `partition_type` for wrapper, but `build_segmentation_model` function get `model_type`, maybe here is a bug in segmentation?

Reply:
Check the codes out:
```
    partition_config = get_partition_config(deploy_cfg)
    if partition_config is not None:
        partition_type = partition_config.get('type', None)
    else:
        codebase_config = get_codebase_config(deploy_cfg)
        # Default Config is 'end2end'
        partition_type = codebase_config.get('model_type', 'end2end')
```
`partition_type` is applied only when `partition_config` is not None. `model_type` is the primary choice when building models.

Reply:
but in segmentation task, the build_model function accept `model_type` as paramter. this means we should set type in codebase_config when partitition model?
```PYTHON
    deploy_cfg, model_cfg = load_config(deploy_cfg, model_cfg)
    backend = get_backend(deploy_cfg)
    model_type = get_codebase_config(deploy_cfg).get('model_type', 'end2end')

    backend_segmentor = __BACKEND_MODEL.build(
        dict(
            type=model_type,
            backend=backend,
            backend_files=model_files,
            device=device,
            deploy_cfg=deploy_cfg,
            data_preprocessor=data_preprocessor,
            **kwargs))
```

Reply:
`model_type` should always be the primary choice as the name **model type** denotes. All you need to do for segmentation is reset the type of `model_type` in `codebase_config` in *configs/mmseg/segmentation_vacc-fp16_static_512x512.py* and register a mmseg model just like you did in mmdet *mmdeploy/codebase/mmdet/deploy/object_detection_model.py*.
```python
codebase_config = dict(model_type='vacc_seg')
```

Reply:
Or, there is another choice. Updating the codes in `build_segmentation_model` if you prefer. And you can use the partition type in mmseg then.